vllm/utils.py

```diff
@@ -746,7 +746,8 @@ def _cuda_device_count_stateless(
     if not torch.cuda._is_compiled():
         return 0
     # bypass _device_count_nvml() if rocm (not supported)
-    nvml_count = -1 if torch.version.hip else torch.cuda._device_count_nvml()
+    nvml_count = torch.cuda._device_count_amdsmi() if (
+        torch.version.hip) else torch.cuda._device_count_nvml()
```

Comment:
can we use `is_hip` (in `utils.py`) instead?

Reply:
We could, I'm following what I thought was a convention for this file specifically: the `find_nccl_library` function also uses bare `torch.version.hip` and `torch.version.cuda`